src/download.rs

```diff
@@ -282,17 +278,21 @@ pub fn image_copy_default(
     // sparse-copy the image, falling back to non-sparse copy if hardware
     // acceleration is unavailable.  But BLKZEROOUT doesn't support
     // BLKDEV_ZERO_NOFALLBACK, so we'd risk gigabytes of redundant I/O.
-    copy(source, &mut dest).chain_err(|| "decoding and writing image")?;
+    //
+    // Amortize write overhead.  The decompressor will produce bytes in
+    // whatever chunk size it chooses.
+    let mut buf_dest = BufWriter::with_capacity(BUFFER_SIZE, dest);
+    copy(source, &mut buf_dest).chain_err(|| "decoding and writing image")?;
+    // we can't into_inner().chain_err() because of lifetime issues
```

Comment:
If I understood this comment correctly, I think you should be doing something like `.into_inner().map_err(|_| ... )` instead (i.e. no point in bubbling up the bufwriter on error, a human message is enough).